hardinfo/dmi_util.c

```diff
@@ -0,0 +1,152 @@
+/*
+ *    HardInfo - Displays System Information
+ *    Copyright (C) 2003-2017 Leandro A. F. Pereira <leandro@hardinfo.org>
+ *    This file
+ *    Copyright (C) 2017 Burt P. <pburt0@gmail.com>
+ *
+ *    This program is free software; you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, version 2.
+ *
+ *    This program is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with this program; if not, write to the Free Software
+ *    Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA
+ */
+
+#include "hardinfo.h"
+#include "dmi_util.h"
+
+const char *dmi_sysfs_root() {
+    char *candidates[] = {
```

Comment:
These tables should always be `static const char *candidates[]`. However, for such a small table, it might be better to just write the function like so:

```C
static const char *dmi_sysfs_root(void)
{
        if (!access("/sys/devices/virtual/dmi", F_OK))
              return "/sys/devices/virtual/dmi";
        if (!access("/sys/class/dmi", F_OK))
              return "/sys/class/dmi";
        return NULL;
}
```